### PR TITLE
mon: osdmonitor: trigger an immediate propose if any newly down osd i…

### PR DESCRIPTION
**Pull request by xiexingguo**
…s detected during tick()

Signed-off-by: xie xingguo <xie.xingguo@zte.com.cn>

### CODE REVIEW

File: src/mon/OSDMonitor.cc [hunk 1]
```diff
@@ -1646,6 +1649,8 @@ bool OSDMonitor::check_failure(utime_t now, int target_osd, failure_info_t& fi)
 
   utime_t grace = orig_grace;
   double my_grace = 0, peer_grace = 0;
+  if (propose_now)
+    *propose_now = false;
```
Comment (liewegas) [inline src/mon/OSDMonitor.cc]: hmm, I woudl drop this, and the weird condition when passing the arg or NULL above, and then only set the value to true

File: src/mon/OSDMonitor.cc [hunk 2]
```diff
@@ -2591,7 +2600,8 @@ void OSDMonitor::tick()
   utime_t now = ceph_clock_now(g_ceph_context);
 
   // mark osds down?
-  check_failures(now);
+  if(check_failures(now))
```
Comment (liewegas) [inline src/mon/OSDMonitor.cc]: s/if(/if (/

File: src/mon/OSDMonitor.cc (hunk 1 above)
Comment (liewegas) [inline src/mon/OSDMonitor.cc]: and hopefully this is the only caller and we can make the argument required instead of optional?

Comment (xiexingguo) [inline src/mon/OSDMonitor.cc]: @liewegas  Actually check_failure() will also be called by prepare_failure(), that is why I try to make this argument optional.

**Comment by xiexingguo** [comment]
Others are fixed.

**Pull request closed without merging**